src/bowling.js

```diff
@@ -0,0 +1,89 @@
+import Frame from './frame.js';
+
+export default class Bowling {
+  constructor() {
+    this.frames = [];
+    this.gameStatus = true;
+    this.totalScore = 0;
+    this._newFrame();
+  }
+
+  roll(pins) {
+    if (!this.gameStatus) return;
+    this.currentFrame.roll(pins, this._isBonus);
+    this.totalScore += pins;
+    this._checkGameStatus();
+    this._prepareNextRoll();
+  }
+
+  _prepareNextRoll() {
+    if (this._isNextFrame) {
+      this._adjustScoresForPastFrames();
+      this._newFrame();
+    }
+    if (this._isBonus) {
```

Comment:
I think the bonus functionality could be moved to the Frame class. The Frame can keep track of whether it still needs to be bonus rolls or not, and decide to count them in or not.
This way this class here can just distribute new rolls to all frames, or just to the last three frames for example, and the frame can do their own calculation.